### PR TITLE
PRM-1112 JS Route: 500 error redirected to error screen

### PR DESCRIPTION
**Pull request by ruthdas03**

### CODE REVIEW

File: app/assets/javascripts/upload/multiFileUpload.js [hunk 1]
```diff
@@ -609,7 +610,13 @@ export class MultiFileUpload {
             method: 'POST',
             body: formData
         })
-            .then(response => response.json())
+            .then(response => {
+                              if (response.status == 500) {
+                              window.location.href = this.getErrorServiceUrl();
+                              } else {
+                                return response.json();
+                              }
```
Comment (LewisMitchell) [inline app/assets/javascripts/upload/multiFileUpload.js:618]: It'd be nice to have some tests for this scenario in the JS spec

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Could you also rebase as well please?

File: app/assets/javascripts/upload/multiFileUpload.js (hunk 1 above)
Comment (ruthdas03) [reply app/assets/javascripts/upload/multiFileUpload.js:618]: Unfortunately, the attempts made to write a test scenario for this din't work. Looks like the 'fetch' method is not properly called/mocked. 
One of such unsuccessful attempts: 
` describe("When upscan error response is 500", () => {
            const errorResponse = new Response(JSON.stringify({}), { status: 500, statusText: 'Internal Server Error', });
                beforeEach((done) => {
                            instance = new MultiFileUpload(container);
                            spyOn(instance, 'setDuplicateInsetText').and.returnValue(Promise.resolve({}));
                            spyOn(instance, 'requestProvisionUpload').and.callThrough((file));
                            spyOn(instance, 'fetch').and.resolveTo(errorResponse);
                            instance.init();
                        });
                it('statusCode=500', (done) => {
                expect(instance.requestProvisionUpload).toHaveBeenCalled();
                done();
                });
            });` 

Please share if there's any reference for such test case.

Comment (ruthdas03) [reply app/assets/javascripts/upload/multiFileUpload.js]: Thanks 😃 , added the test.

**Comment by LewisMitchell** [comment]
Could you rebase please?

**Comment by LewisMitchell** [comment]
I think your local machine may be behind slightly. Could you `git fetch` and rebase again as it says this branch is 3 commits behind

**Review by LewisMitchell — APPROVED** [review]

**Review by AndrewRButcher — APPROVED** [review]

**Pull request merged**